project/recursive_sorting.py

```diff
@@ -45,33 +45,60 @@ def merge_sort_in_place(arr, l, r):
 
 
 # TO-DO: implement the Quick Sort function below USING RECURSION
-def quick_sort( arr, low, high ):
-        # base case
-    if high-low <= 0:
-        return arr
-
-    pivot = low
-    for i in range(low + 1, high + 1):
-        # print("This is I", [i])
-        # print(arr)
-        if arr[i] < arr[pivot]:
-            # move to LHS of pivot - 2 swaps
-            temp = arr[i]
-            arr[i] = arr[pivot + 1]
-            arr[pivot + 1] = temp
-            # print(arr)
-            # print("This is I after move LHS of pivot 1 swap", [i])
-            temp = arr[pivot]
-            arr[pivot] = arr[pivot + 1]
-            arr[pivot + 1] = temp
-            pivot += 1
-            print(arr)
-            # print("This is I after move LHS of pivot 2 swaps", [i])
-
-            #path to the base case?
-            #Call quick_sort
+# def quick_sort( arr, low, high ):
+#         # base case
+#     if high-low <= 0:
+#         return arr
 
-    return arr
+#     pivot = low
+#     for i in range(low + 1, high + 1):
+#         # print("This is I", [i])
+#         # print(arr)
+#         if arr[i] < arr[pivot]:
+#             # move to LHS of pivot - 2 swaps
+#             temp = arr[i]
+#             arr[i] = arr[pivot + 1]
+#             arr[pivot + 1] = temp
+#             # print(arr)
+#             # print("This is I after move LHS of pivot 1 swap", [i])
+#             temp = arr[pivot]
+#             arr[pivot] = arr[pivot + 1]
+#             arr[pivot + 1] = temp
+#             pivot += 1
+#             print(arr)
+#             # print("This is I after move LHS of pivot 2 swaps", [i])
+
+#             #path to the base case?
+#             #Call quick_sort
+
+#     return arr
+
+from random import randint
+
+#create a randomized, unsorted array for testing
+def create_array(size=10, max=50):
+  return [randint(0, max) for _ in range(size)]
+
+def quicksort(a):
```

Comment:
nice! we talked about this already, but great job getting this to work with a simple array.
remember that this takes up more memory though, but you also leave you original array untouched and you `return` a new copy of the array that is sorted.